### DISCUSSION TITLE
Memoria - Speedrun category

### DISCUSSION THREAD

**Discussion by Tirlititi** [Albeoris/Memoria#210] [Ideas]
There has been discussions in the speedrunning community for having a category in which the Memoria mod and Moguri mod are allowed. A speedrun category cannot be allowed to change its rules or settings every morning, if ever. So that requires the community to agree on these rules and the mod version that would be used for every speedrunner.

I open this discussion topic in order to discuss about what could be a speedrun version of Memoria.

In terms of the organisation related to Github, I guess the best way is to create a branch for it.

In terms of features that would need to be added to Memoria, I have been told the followings:
- **Either** prevent any external customisation (in particular, ``Memoria.ini`` configurations would be fixed)
- **or** have a checksum system that would give all the necessary informations about the settings used for playing, and that could not be faked
- Some tweaks to how inputs work, to bring them more in line with the PSX version
- Add the button Square as an input to close dialog boxes (in the PSX version, both buttons Cross and Square close dialog boxes)
- Make it so the cursor is able to wrap from top to bottom and vice versa
- Disable mouse inputs as a control method (I guess for both the movements on the field and the dialogs/menus)
- Possibly have an option to always display the menu timer

Maybe, there are some features of Memoria that should be removed as well because the speedrunning scene doesn't play with them. Maybe I should revert the changes I made when working on the SFX Rework system, as it has a small impact on the flow of battles overall. Maybe having the game run at 60 FPS would not be good for everyone for streaming, or maybe on the contrary that's a feature they'd want to keep.
I don't know either if they want to keep the [boss skip exploit](https://www.speedrun.com/ff9/thread/rjo4x). Actually, I don't even know if that exploit is still possible with the latest version and the FPS manager.

Memoria provides so many features and bug fixes... I am not exactly sure of which ones are important to the speedrunning community, so I would also like this discussion page to be a place where we answer questions about the differences between the vanilla game and the Memoria-modded game.

**Comment by Dubldrop** [comment]
Its been a while since any discussion has really happened so I want to give an update on my end

I get the feeling that while this project is definitely going to be welcomed by the community, their understanding of what is and isnt possible, as well as what already is and isnt implimented in the base engine is leading to a lack of constructive dialogue with the general community. With that in mind, I think the best way to proceed is to get a rough product, a working beta, ready and able to be viewed for constructive feedback

With that in mind, I have a rough outline of what we would need to accomplish before I can showcase it for the rest of the speedrunning community:

1) High Priority
Id like to start with implimenting some QoL features. Most of this is graciously already done, but I will list the features I would hope to see added:
 - The ability to progress dialogue with the Square(Card Battle) key, in line with PSX
 - The ability to wrap the cursor in battle menues
 - a tweak to how shoulder scrolling in menues works to revert to PSX style: L/R scrolls the page, cursor lands on the same line on the next page (HD default scrolls the next page and lands the cursor on the bottom or top respective to direction)
 - The ability to disable mouse inputs entirely
 -  An easy to access configuration file with the default character names for easy renaming before loading the game (This would allow custom names if using the disable naming screen feature)

2) Medium Priority
I think some of the changes I proposed in section 1 will likely need some accompanying graphic work to really polish. There are a ton of talented graphic designers in the speedrun community I can outsource this to, so the main work here is knowing what would need to be done in order to incorporate it.

 
3) Low Priority
Given the nature and desire for different categories and available time to play them, I think it would be wise to consider what would be the default configurable settings for the speedrun. I can envision having multiple "presets" depending on the category one wishes to run. For example, 1 preset that aims to emulate the PSX experience as faithfully as possible (reverting mechanics like reflect all damage on a single enemy, adjusting the encounter rate, ect), another to have a more HD experience with the additional QoL changes, and another for a shorter experience coupled with something like the cutscene remover. I would propose working towards having a launcher setting or other custom launch config that can automate the changing of the needed settings to the chosen ones for each category. 

I am more than willing to test out different configuration setups to determine a preset, and then tweak those based on community feedback. Laying the groundwork for this system is the challenging part, once its there, I can test the different settings and work with the community to finalize values

And of course the ability to checksum the version, and lock out most configureable files is the last thing needed as far as making this viable in terms of leaderboards

So far most of the bugfixes Memoria has incorporated dont really have an effect on the speedrun, so I think its pretty okay to keep them in place. "The fact that you fixed the music bug during alexandria escape is reason by itself to want to use this version for speedruns" -Cease. The main concern we have are changes to the actual mechanics of the game (Battle Mechanics, stuff like that.). And the way they have been implemented in Memoria is ideal, so long as we can block access to tweaking them, or at least make tweaking them change a validation checksum.
